### PR TITLE
Add functions useful for interacting with docker images and building remote images

### PR DESCRIPTION
**Pull request by yorinasub17**
This PR introduces a few functions that are useful for interacting with docker images, and building remote images:

- `ListImages` for listing images in the docker daemon
- `DeleteImage` for cleaning up images in the docker daemon
- `DoesImageExist` for checking whether a given image exists in the daemon
- `GitCloneAndBuild` for building a docker image from a remote git repo

### CODE REVIEW

File: modules/docker/build_test.go [hunk 1]
```diff
@@ -60,3 +64,23 @@ func TestBuildWithTarget(t *testing.T) {
 	out := Run(t, tag, &RunOptions{Remove: true})
 	require.Contains(t, out, text1)
 }
+
+func TestGitCloneAndBuild(t *testing.T) {
```
Comment (denis256) [inline modules/docker/build_test.go:68]: `t.Parallel()` was removed by intention?

**Review by denis256 — DISMISSED** [review]
I was wondering if will make sense to add an additional function to explicitly pull image, it may be helpful in cases when we have `latest` tag and want to avoid caching locally.

File: modules/docker/build_test.go (hunk 1 above)
Comment (yorinasub17) [reply modules/docker/build_test.go:68]: Ah good catch! Was not intentional. Added in! [443f3f8](https://github.com/gruntwork-io/terratest/pull/1062/commits/443f3f807fd24fcf28c2cdf3c1cd35831cafcf7d)

**Review by denis256 — APPROVED** [review]

**Comment by yorinasub17** [comment]
Thanks for review! Build failures are unrelated to this change, so will go ahead and merge this in.

**Pull request merged**